src/main/java/onlinebookstore/validation/IsbnValidator.java

```diff
@@ -0,0 +1,14 @@
+package onlinebookstore.validation;
+
+import jakarta.validation.ConstraintValidator;
+import jakarta.validation.ConstraintValidatorContext;
+import java.util.regex.Pattern;
+
+public class IsbnValidator implements ConstraintValidator<Isbn, String> {
+    private static final String ISBN_PATTERN = "^\\d{2}-\\d{3}-\\d{3}-\\d{2}$";
```

Comment:
you can use `Pattern` directly:
```suggestion
    private static final Pattern ISBN_PATTERN = Pattern.compile("^\\d{2}-\\d{3}-\\d{3}-\\d{2}$");
```